controllers/authentication/openidconnect_controller_test.go

```diff
@@ -404,6 +404,109 @@ var _ = Describe("OpenIDConnect controller", func() {
 				Expect(resp).To(BeNil())
 			})
 		})
+
+		Context("Extra claims handling", func() {
+			unionHandler := newUnionAuthTokenHandler()
+			issuer1URL := "https://issuer1"
+			handler1 := &mockAuthRequestHandler{returnUser: user1, isAuthenticated: true, issuerURL: issuer1URL}
+			handler2 := &mockAuthRequestHandler{returnUser: user2, isAuthenticated: false, issuerURL: "https://issuer2"}
+
+			authUID := uuid.NewUUID()
+			unionHandler.registerHandler(issuer1URL, "1", &authenticatorInfo{
+				Token: handler1,
+				name:  "1",
+				uid:   authUID,
+			})
+			unionHandler.registerHandler("https://issuer2", "2", &authenticatorInfo{
+				Token: handler2,
+				name:  "2",
+				uid:   uuid.NewUUID(),
+			})
+
+			claims := map[string]interface{}{
+				"iss":    issuer1URL,
+				"claim1": "value1",
+				"claim2": 2,
+				"claim3": []interface{}{
+					"value3",
+					3,
+				},
+			}
+
+			getIssuer1AuthenticatorInfo := func() *authenticatorInfo {
+				unionHandler.mutex.Lock()
+				defer unionHandler.mutex.Unlock()
```

Comment:
I think that this does not guard against any race conditions. If this is what you are trying to achieve when setting the `extraClaims` then something like this is more appropriate.

```golang
setIssuer1ExtraClaims := func(extra []string) {
    unionHandler.mutex.Lock()
    defer unionHandler.mutex.Unlock()
    unionHandler.issuerHandlers[issuer1URL]["1"].extraClaims = extra
}
```

Reply:
Thank you for the suggestion, it's indeed way cleaner.